enacts/calc.py

```diff
@@ -619,6 +620,103 @@ def daily_tobegroupedby_season(
     return daily_tobegroupedby_season
 
 
+def assign_season_coords(
+    time_series,
+    start_day,
+    start_month,
+    end_day,
+    end_month,
+    time_dim="T",
+):
+    """Assign season start and season end coordinates to this object,
+    mapping time dimension dates that belong to a season to their season start and end.
+    Data outside seasons are dropped.
+
+    Parameters
+    -----------
+    time_series : DataArray, Dataset
+        Time-dependent data to be grouped.
+    start_day : int
+        Day of the start date  of the season.
+    start_month : int
+        Month of the start date of the season.
+    end_day : int
+        Day of the end date of the season.
+    end_month : int
+        Day of the end date of the season.
+    time_dim : str, optional
+        Time coordinate in `time_series` (default `time_dim` ="T").
+
+    Returns
+    -------
+    time_series : DataArray, Dataset
+        `time_series` where days outside the seasons of interest are dropped
+        and with two additional coordinates `season_start` and `season_end`
+        that map the time dimension dates that belong to a season to
+        their season start and end respectively.
+
+    See Also
+    --------
+    xarray.DataArray.assign_coords, xarray.DataArray.groupby
+
+    Notes
+    -----
+    If the first (last) season is truncated, because `time_series` starts (ends) within it,
+    The first (last) element of `season_start` ( `season_end` ) differs from all others
+    and is the first (last) element of `time_dim` .
```

Comment:
If the caller wants only complete seasons, how hard will it be to check whether the first and last seasons are complete, and drop them if not? It seems complicated. Maybe this function should have a boolean option to drop  or include them?

Reply:
I would be in favor of letting the user slice (or pad) to make sure they get complete seasons. It's trivial enough not to build it inside this function. 
Independently, it would be nice anyways to have an Ingrid-like minfrac option but I feel that this should rather be a parameter of the operator that is applied to the labeled days-in-season... But maybe I feel that because that's how it is in Ingrid. Here, we are going to apply xarray operators already written that we are not going to rewrite, so putting the minfrac parameter at this level would insure it's used downstream. Maybe that's what we should do? Then it would probably make sense not to truncate the potentially incomplete tailing seasons but pad them with missing and let minfrac decide if they are kept or dropped.

Reply:
Padding with NaNs is an interesting potential solution, but I'm not sure about it. It would obscure the difference between "the dataset doesn't go back that far" and "the dataset goes back that far and contains NaNs."

Adding a minfrac option seems like a good idea, but we don't have to do it now. We can add it later, as long as it doesn't change the default behavior. That's probably the way to go.

Reply:
My suggestion was the user pads (or slices) before applying assign_season_coords so they are in the know. If they voluntarily choose to apply seasons thresholds on data that start or end within them (without slicing/padding before to avoid it), then they get "smaller" seasons where that happens (the first and/or the last season). And that is also unambiguous that those special seasons come from the fact that there were no data in the first place those days. And such information is actually carried over so it's arguably a good behavior.

As for a future minfrac, I guess we could drop the groups that don't comply. Default behavior would be minfrac = 0 (ie you get a group as long as there is 1 data point). Then at present, if there is no data points at all in the first place in a group, there is no data points to assign coord to so it's already dropped in a way. If there are data points but they all are NaN, assign_season_coords will assign start/end_season coords to the points in such a group. Then the returning behavior would be the behavior of the function applied to groupby.

So maybe we should drop data points where all group members are NaN? to be consistent in introducing minfrac in the future? Or else we do nothing now but when we introduce minfrac, we set back to NaN all the data values that belong to a group that doesn't comply to minfrac.